File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/service/EmailService.java
package com.windea.demo.cloudcollect.core.service;

/**
 * 邮件的服务。
 */
public interface EmailService {
	/**
	 * TODO 发送邮件。
	 */
	void sendEmail();

	/**
	 * TODO 发送激活邮件。
	 */
	void sendActivateEmail();

	/**
	 * TODO 发送重置密码邮件。
	 */
	void sendResetPasswordEmail();

	/**
	 * TODO 发送欢迎邮件。
	 */
	void sendHelloEmail();
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/service/EmailService.kt
package com.windea.demo.cloudcollect.core.service

import com.windea.demo.cloudcollect.core.domain.entity.*

interface EmailService {
	/**发送激活邮件。*/
	fun sendActivateEmail(user: User, activateCode: String)
	
	/**发送欢迎邮件。*/
	fun sendHelloEmail(user: User)
	
	/**发送重置密码邮件。*/
	fun sendResetPasswordEmail(user: User, resetPasswordCode: String)
	
	/**发送重置密码成功邮件。*/
	fun sendResetPasswordSuccessEmail(user: User)
}
